lib/OpenQA/WebAPI/Controller/Step.pm

```diff
@@ -533,10 +535,11 @@ sub save_needle_ajax {
         }
         $self->emit_event('openqa_needle_modify',
             {needle => "$baseneedle.png", tags => $json_data->{tags}, update => 0});
-        my $info = {info => "Needle $needlename created/updated."};
-        if ($job->can_be_duplicated) {
-            $info->{restart} = $self->url_for('apiv1_restart', jobid => $job->id);
-        }
+        my $info = {
+            info            => "Needle $needlename created/updated",
+            interactive_job => ($job->worker_id && $job->worker->get_property('INTERACTIVE') ? $job->id : undef),
```

Comment:
please write your statements so they fit in this review window without scrolling.